src/client/dfuse/il/int_posix.c

```diff
@@ -822,7 +822,8 @@ check_ioctl_on_open(int fd, struct fd_entry *entry, int flags, int status)
 
 	cont->ioc_open_count += 1;
 
-	pthread_mutex_unlock(&ioil_iog.iog_lock);
+	rc = pthread_mutex_unlock(&ioil_iog.iog_lock);
+	D_ASSERT(rc == 0);
```

Comment:
https://github.com/daos-stack/daos/blob/master/src/client/dfuse/il/int_posix.c#L836-L837
assert is used in the following code in this function. I guess it may be fine. The failure of pthread_mutex_unlock may indicates something is wrong.

Reply:
pthread_mutex_unlock() can only fail if it's called incorrectly so I think assert is OK here.